main.js

```diff
@@ -0,0 +1,188 @@
+element. It envokes the setNames function.
+*/
+firstNameHolder.addEventListener('blur', function() {
+    setNames(firstName, lastName);
+});
+/*This is the event listener to listen for the blur event on the lastNameHolder
+element. It envokes the setNames function.
+*/
+lastNameHolder.addEventListener('blur', function() {
+    setNames(firstName, lastName);
+});
+/*This is the setNames function that takes the input on the element and sets 
+the first character to upper case.
+*/
+function setNames(fName, lName){
+    let fName = document.getElementById('firstName');
+    let lName = document.getElementById('lastName');
+    fName.charAt(0).toUpperCase();
+    fName.charAt(fName.length).toLowerCase();
+    lName.charAt(0).toUpperCase();
+    lName.charAt(lName.length).toLowerCase();
+}
+/*This is the event listener for the ageHolder element. It checks the value
+of the age input and sends an error message if a non number value has been 
+entered. It then sets the attribute back to empty and refocuses the cursor.
+*/
+ageHolder.addEventListener('blur' , function(){
+    if(Number.isNaN(age.value)){
+        alert('Please input only a number for age');
+        age.setAttribute('input', '');
+        age.focus();
+    }
+});
+/*These statements assign the element errorHolder to the variable errorHolder, removes
+the class of 'visible' from the errorHolder class list then removes any children from the
+errorHolder element if any.
+*/
+const errorHolder = document.getElementById('errorHolder');
+errorHolder.classList.remove('visible');
+while(errorHolder.firstChild) {
+    errorHolder.removeChild(errorHolder.firstChild);
+}
+/*This is an event listener for missing field errors that creates an ul of errors to 
+display if any errors occur using the buildErrorList function and applies the errorBox 
+class to the error message box.
+*/
+form.addEventListener('submit', function(e) {
+    const errors = [];
+    if(firstName.value == ''){
+        errors.push('First Name cannot be blank.');
+    }
+    if(lastName.value == ''){
+        errors.push('Last name cannot be blank.');
+    }
+    if(emailAddress.value == ''){
+        errors.push('E-Mail address cannot be blank.');
+    }
+    if(age.value == ''){
+        errors.push('Age cannot be blank.');
+    }
+    if(meal.value == ''){
+        errors.push('You must select a meal.');
+    }
+    if(mealOption.value == ''){
+        errors.push('You must select what you ate.');
+    }
+    if(mealRating.value == ''){
+        errors.push('You must rate your meal.');
+    }
+    if(errors.length > 0){
+        errorHolder.appendChild(buildErrorList());
+        errorHolder.classList.add('errorBox');
+        errorHolder.classList.add('visible');
+        e.preventDefault();
+    }
+});
+
+//This is the buildErrorList function to create the ul of errors li and appends it.
+function buildErrorList() {
+    for(let error of errors){
```

Comment:
This gives you an error because errors is defined within the listener - scoped for that listener. Now you are calling it outside its scope so JS doesn't know it exists.